src/lumigo_tracer/auto_tag/auto_tag_event.py

```diff
@@ -0,0 +1,60 @@
+import os
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+from lumigo_tracer.user_utils import add_execution_tag
+from lumigo_tracer.parsers.utils import str_to_list
+from lumigo_tracer.utils import get_logger
+
+
+AUTO_TGA_API_GW_HEADERS = str_to_list(os.environ.get("LUMIGO_AUTO_TGA_API_GW_HEADERS", "")) or []
+
+
+class EventAutoTagHandler(ABC):
+    @staticmethod
+    @abstractmethod
+    def is_supported(event) -> bool:
+        raise NotImplementedError()
+
+    @staticmethod
+    @abstractmethod
+    def auto_tag(event) -> Dict:
+        raise NotImplementedError()
+
+
+class ApiGWHandler(EventAutoTagHandler):
+    @staticmethod
+    def is_supported(event) -> bool:
+        if (
+            len(AUTO_TGA_API_GW_HEADERS) > 0
+            and isinstance(event, Dict)  # noqa
+            and event.get("requestContext")  # noqa
+            and event.get("requestContext", {}).get("domainName")  # noqa
+            and event.get("requestContext")  # noqa
+            and event.get("requestContext", {}).get("requestId")  # noqa
+        ):
+            return True
+        return False
+
+    @staticmethod
+    def auto_tag(event):
+        headers = event.get("headers", [])
+        for key in AUTO_TGA_API_GW_HEADERS:
+            if key in headers:
+                add_execution_tag(key, headers[key])
+
+
+class AutoTagEvent:
+    @staticmethod
+    def auto_tag_event(event: Dict = None, handlers: List[EventAutoTagHandler] = None):
```

Comment:
```suggestion
    def auto_tag_event(event: Optional[Dict] = None, handlers: Optional[List[EventAutoTagHandler]] = None) -> None:
```
`None` isn't the same type as `dict`. We need to say `Union[None, dict]` or the shorter version: `Optional[dict]`.